components/SearchBar.tsx

```diff
@@ -21,12 +26,26 @@ export default function SearchBar({
         type="text"
         placeholder="Search for your favorite records"
         className="grow pl-10 py-2 rounded-xl text-black"
-        onChange={(e) => setUserInput(e.target.value)}
+        onChange={(e) => onInput(e.target.value)}
+        value={searchValue ? searchValue : ""}
```

Comment:
in this case searchValue being an empty string makes more sense imo. Simpler, no turneries, etc.